packages/components/src/VirtualizedList/CellTitle/CellTitle.component.js

```diff
@@ -49,12 +49,24 @@ class CellTitle extends React.Component {
 
 		return (
 			<div id={titleId} className={classNames('tc-list-title', theme['tc-list-title'])}>
-				{iconKey && rowData[iconKey] && <Icon name={rowData[iconKey]} className={theme.icon} />}
+				{iconKey &&
+					rowData[iconKey] && (
+						<Icon
+							name={rowData[iconKey]}
+							className={classNames(theme.icon, {
+								[theme['tc-main-title-icon-active']]: onClick,
+								'tc-main-title-icon-active': onClick,
```

Comment:
You don't need to add specific conditional classnames on each element. Just put one on the main div, and in the scss, you can target existing classnames on each part
```
.my-class {
	...

	&:hover,
	&:focus {
		.icon: {}
		.main-title {}
	}
}
```

Let's not name it `active` but something with filter-on-hover for example, something more meaningful. Active means something else in css.

Reply:
OK i change the name but i need to add some condition. this css is only active when onClick is defined.

Reply:
In you example, the hover, focus is always applied when onClick exists or not.

Reply:
No because you scope it with `my-class`. This className is on the component if you have the onClick